src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/SerializationContextTests.cs

```diff
@@ -0,0 +1,268 @@
+            json = JsonSerializer.Serialize(myType2, DefaultContext.MyType2);
+            myType2 = JsonSerializer.Deserialize(json, ((ITestContext)MetadataContext.Default).MyType2);
+            AssertFastPathLogicCorrect(json, myType2, DefaultContext.MyType2);
+        }
+
+        [Fact]
+        public override void SerializeObjectArray()
+        {
+            IndexViewModel index = CreateIndexViewModel();
+            CampaignSummaryViewModel campaignSummary = CreateCampaignSummaryViewModel();
+
+            string json = JsonSerializer.Serialize(new object[] { index, campaignSummary }, DefaultContext.ObjectArray);
+            object[] arr = JsonSerializer.Deserialize(json, ((ITestContext)MetadataContext.Default).ObjectArray);
+
+            JsonElement indexAsJsonElement = (JsonElement)arr[0];
+            JsonElement campaignSummeryAsJsonElement = (JsonElement)arr[1];
+            VerifyIndexViewModel(index, JsonSerializer.Deserialize(indexAsJsonElement.GetRawText(), ((ITestContext)MetadataContext.Default).IndexViewModel));
+            VerifyCampaignSummaryViewModel(campaignSummary, JsonSerializer.Deserialize(campaignSummeryAsJsonElement.GetRawText(), ((ITestContext)MetadataContext.Default).CampaignSummaryViewModel));
+        }
+
+        [Fact]
+        public override void SerializeObjectArray_WithCustomOptions()
+        {
+            IndexViewModel index = CreateIndexViewModel();
+            CampaignSummaryViewModel campaignSummary = CreateCampaignSummaryViewModel();
+
+            ITestContext context = SerializationContextWithCamelCase.Default;
+            Assert.Same(JsonNamingPolicy.CamelCase, ((JsonSerializerContext)context).Options.PropertyNamingPolicy);
+
+            string json = JsonSerializer.Serialize(new object[] { index, campaignSummary }, context.ObjectArray);
+            object[] arr = JsonSerializer.Deserialize(json, ((ITestContext)MetadataContext.Default).ObjectArray);
+
+            JsonElement indexAsJsonElement = (JsonElement)arr[0];
+            JsonElement campaignSummeryAsJsonElement = (JsonElement)arr[1];
+
+            ITestContext metadataContext = new MetadataContext(new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+            VerifyIndexViewModel(index, JsonSerializer.Deserialize(indexAsJsonElement.GetRawText(), metadataContext.IndexViewModel));
+            VerifyCampaignSummaryViewModel(campaignSummary, JsonSerializer.Deserialize(campaignSummeryAsJsonElement.GetRawText(), metadataContext.CampaignSummaryViewModel));
+        }
+
+        [Fact]
+        public override void SerializeObjectArray_SimpleTypes_WithCustomOptions()
+        {
+            JsonSerializerOptions options = new() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+            ITestContext context = new SerializationContext(options);
+
+            string json = JsonSerializer.Serialize(new object[] { "Hello", "World" }, typeof(object[]), (JsonSerializerContext)context);
```

Comment:
There's no check against whether the JSON was written with camel case.

Reply:
Added a test for this elsewhere. In this specific case, we have an array and the policy isn't applied to collection elements.